ui/src/app/pages/data/data-set/export-dataset-modal.spec.tsx

```diff
@@ -492,4 +503,213 @@ describe('ExportDatasetModal', () => {
       })
     );
   });
+
+  it('Dataset copied to clipboard when "copy code" button is clicked', async () => {
+    // Arrange
+    datasetApiStub.codePreview = {
+      text: '<div id="notebook">print("hello world!")</div>',
+    };
+
+    // Act
+    component(testProps);
+    await user.click(findCopyCodeButton());
+
+    // Assert
+    const clipboardText = await navigator.clipboard.readText();
+    expect(clipboardText).toBe(
+      '<div id="notebook">print("hello world!")</div>'
+    );
+    // Assert tooltip is shown
+    await screen.findByText('Dataset query copied to clipboard');
+  });
+
+  it('Dataset copied to clipboard when "copy code" icon button is clicked in expanded view', async () => {
+    // Arrange
+    datasetApiStub.codePreview = {
+      text: '<div id="notebook">print("hello world!")</div>',
+    };
+
+    // Act
+    component(testProps);
+    await waitUntilDoneLoading();
+
+    clickShowCodePreviewButton();
+
+    await waitUntilDoneLoading();
+    await user.click(screen.getByLabelText('Dataset query copy icon button'));
+
+    // Assert
+    const clipboardText = await navigator.clipboard.readText();
+    expect(clipboardText).toBe(
+      '<div id="notebook">print("hello world!")</div>'
+    );
+    // Assert tooltip is shown
+    await screen.findByText('Dataset query copied to clipboard');
+  });
+
+  it('Nothing should be clickable when notebooks are loading', async () => {
+    testProps.dataset.prePackagedConceptSet = [
```

Comment:
is this needed?

Reply:
I believe so. Without this, the genomics tools radio buttons would not show up.

Reply:
ok, got it